crates/masonry/src/widget/button.rs

```diff
@@ -82,13 +82,18 @@ impl Widget for Button {
                 }
             }
             PointerEvent::PointerUp(_, _) => {
-                if ctx.is_active() && !ctx.is_disabled() {
+                if ctx.is_active() && ctx.is_hot() && !ctx.is_disabled() {
                     ctx.submit_action(Action::ButtonPressed);
                     ctx.request_paint();
                     trace!("Button {:?} released", ctx.widget_id());
                 }
                 ctx.set_active(false);
             }
+            PointerEvent::PointerLeave(_) => {
+                // If the screen was locked whilst holding down the mouse button, we don't get a `PointerUp`
+                // event, but should no longer be active
```

Comment:
This doesn't need checking but since I often switch between having 1 or 2 external monitors attached to my laptop, I wonder if `winit` handles a `PointerLeave` in that scenario as well!